node/malus/src/variants/suggest_garbage_candidate.rs

```diff
@@ -88,131 +92,166 @@ where
-									.expect("channel is still open");
-							},
-							_ => {
-								panic!("Unable to fetch validation data");
-							},
-						}
-					}),
-				);
-
-				let (validation_data, validation_code, n_validators) = receiver.recv().unwrap();
-
-				let validation_data_hash = validation_data.hash();
-				let validation_code_hash = validation_code.hash();
-				let validation_data_relay_parent_number = validation_data.relay_parent_number;
-
-				gum::trace!(
+				gum::debug!(
 					target: MALUS,
 					candidate_hash = ?candidate.hash(),
 					?relay_parent,
-					?n_validators,
-					?validation_data_hash,
-					?validation_code_hash,
-					?validation_data_relay_parent_number,
-					"Fetched validation data."
+					"Received request to second candidate",
 				);
 
-				let malicious_available_data =
-					AvailableData { pov: Arc::new(pov.clone()), validation_data };
-
-				let pov_hash = pov.hash();
-				let erasure_root = {
-					let chunks =
-						erasure::obtain_chunks_v1(n_validators as usize, &malicious_available_data)
-							.unwrap();
-
-					let branches = erasure::branches(chunks.as_ref());
-					branches.root()
-				};
-
-				let (collator_id, collator_signature) = {
-					use polkadot_primitives::v2::CollatorPair;
-					use sp_core::crypto::Pair;
-
-					let collator_pair = CollatorPair::generate().0;
-					let signature_payload = polkadot_primitives::v2::collator_signature_payload(
-						&relay_parent,
-						&candidate.descriptor().para_id,
-						&validation_data_hash,
-						&pov_hash,
-						&validation_code_hash,
-					);
+				// Need to draw value from Bernoulli distribution with given probability of success defined by the Clap parameter.
+				// Note that clap parameter must be f64 since this is expected by the Bernoulli::new() function, hence it must be converted.
+				let distribution = Bernoulli::new(self.percentage / 100.0).expect("Invalid probability! Percentage cannot be < 0 or > 100.");
 
-					(collator_pair.public(), collator_pair.sign(&signature_payload))
-				};
-
-				let malicious_commitments =
-					create_fake_candidate_commitments(&malicious_available_data.validation_data);
-
-				let malicious_candidate = CandidateReceipt {
-					descriptor: CandidateDescriptor {
-						para_id: candidate.descriptor().para_id,
-						relay_parent,
-						collator: collator_id,
-						persisted_validation_data_hash: validation_data_hash,
-						pov_hash,
-						erasure_root,
-						signature: collator_signature,
-						para_head: malicious_commitments.head_data.hash(),
-						validation_code_hash,
-					},
-					commitments_hash: malicious_commitments.hash(),
-				};
-				let malicious_candidate_hash = malicious_candidate.hash();
+				// Draw a random value from the distribution, where T: bool, and probability of drawing a 'true' value is = to percentage parameter,
+				// using thread_rng as the source of randomness.
+				let true_or_false = distribution.sample(&mut rand::thread_rng());
```

Comment:
This variable name is not quite fortunate, I'd suggest something like `produce_candidate_sampled`.